plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/model/CodeWhispererModel.kt

```diff
@@ -81,10 +89,19 @@ data class SupplementalContextInfo(
 }
 
 data class RecommendationContext(
-    val details: List<DetailContext>,
+    val details: MutableList<DetailContext>,
     val userInputOriginal: String,
     val userInputSinceInvocation: String,
-    val position: VisualPosition
+    val position: VisualPosition,
+    val jobId: Int,
+    var typeahead: String = "",
+)
+
+data class PreviewContext(
+    val jobId: Int,
+    val detail: DetailContext,
+    val userInput: String,
+    val typeahead: String,
```

Comment:
dumb q, what's the difference between these 2 typeaheads

Reply:
previously we have typeahead and typeaheadoriginal for the purpose of accounting leading spaces in the typeahead, 
for example, typeaheadoriginal will be `     System` and typeahead will be `System`, and then we use typeahead to do the suggestion matching logic. This is to preserve the suggestions as much as possible.

Now we trigger more so we have more display oppotunities, I feel like this becomes a bit unnecessary and makes the logic complicated, so remove typeahead original to make it no longer accomodate leading spaces.